### DISCUSSION TITLE
Get sqitch deploy calling a specific command to load data.

### DISCUSSION THREAD

**Discussion by baugereau** [theory/sqitch#725] [Q&A] [answered]
Hello,

is there a way to add a change that would be performed by a dedicated database client? Depending on SQL INSERT INTO/DELETE FROM /TRUNCATE to (un)load data is an important constraint in case of large objects (e.g. loading XML document) and geospatial objects (gdal/ogr tools). Is there a possibility to indicate to the `sqitch deploy` command to execute a specific command? Note, for such operation there could be a different client for the deploy (dedicated command), and revert and verify (sql scripts).

Thanks !

**Comment by theory** [comment]
No, there is not. The pattern that has emerged to try to address the need for custom clients is to add something like a shell script to your repository that knows what things it must do, and then set the `engine.$name.client` command to point to it. For example:


``` sh
sqitch engine alter pg --client ./bin/psql_wrapper.sh
```

Then teach your script to do the right thing based on some criteria, like parsing the first few lines of each change script, or perhaps knowing to do something for changes with a particular name. Note that the client will be called with Sqitch with the arguments required for your CLI. For Postgres, for example, it assumes you're using `psql`, so passes these options based on the target configuration:

https://github.com/sqitchers/sqitch/blob/aace1e837b5e07fa0d8cfb924505b221f58079c4/lib/App/Sqitch/Engine/pg.pm#L63-L68

As well as these options for every invocation:

https://github.com/sqitchers/sqitch/blob/aace1e837b5e07fa0d8cfb924505b221f58079c4/lib/App/Sqitch/Engine/pg.pm#L91-L96

And then to run a change script, it passes `--file`:

https://github.com/sqitchers/sqitch/blob/aace1e837b5e07fa0d8cfb924505b221f58079c4/lib/App/Sqitch/Engine/pg.pm#L329

So you could write a shell script to parse out the `--file` option and determine if it should call something else.

**Reply by baugereau** [reply]
Thanks for your guidance. I keep the idea and I will try it whenever I will have a more refine data management plan in place and experimented a bit further this question.

**Comment by baugereau** [comment, marked as answer]
thanks for the support !
